src/components/ImageSwitcher.tsx

```diff
@@ -0,0 +1,50 @@
+"use client";
+import Image from "next/image";
+import { useEffect, useState } from "react";
+
+const ImageSwitcher = () => {
+  const [theme, setTheme] = useState("light"); // Default to light
+
+  // Function to update theme based on system preference
+  const updateTheme = (e: MediaQueryListEvent) => {
+    setTheme(e.matches ? "dark" : "light");
+  };
+
+  useEffect(() => {
+    // Check the initial system theme
+    const mediaQuery = window.matchMedia("(prefers-color-scheme: dark)");
+    setTheme(mediaQuery.matches ? "dark" : "light");
+
+    // Listen for changes to the system theme
+    mediaQuery.addEventListener("change", updateTheme);
+
+    // Cleanup the event listener on component unmount
+    return () => {
+      mediaQuery.removeEventListener("change", updateTheme);
+    };
+  }, []);
+
+  return (
+    <div>
+      {theme === "light" ? (
```

Comment:
The logic here can be simplified instead of de duping all the lines use a ternary operator on the prop values that change.

Reply:
```
<Component
  someProp={`common-value ${darkMode ? 'bg-black' : 'bg-white'}`}
/>
```